packages/webextension/index.js

```diff
@@ -0,0 +1,58 @@
+const web = require('@neutrinojs/web');
+const webExt = require('web-ext').default;
+const { join } = require('path');
+
+const MODULES = join(__dirname, 'node_modules');
+
+module.exports = (neutrino, options = {}) => {
+  neutrino.use(web, options);
+
+  if (neutrino.options.debug) {
+    webExt.util.logger.consoleStream.makeVerbose();
+  }
+
+  neutrino.config
+    .output
+      .delete('chunkFilename')
+      .hotUpdateChunkFilename('hot/hot-update.js')
+      .hotUpdateMainFilename('hot/hot-update.json')
+      .end()
+    .devServer
+      .clear()
+      .end()
+    .module
+      .rules
+        .delete('worker')
+        .end()
+      .end()
+    .plugins
+      .delete(options.pluginId || 'html')
+      .end()
+    .resolve
+      .modules
+        .add(MODULES)
+        .end()
+      .end()
+    .resolveLoader
+      .modules
+        .add(MODULES)
+        .end()
+      .end()
+    .when(neutrino.options.command === 'start', () => {
+      webExt.cmd.run({
+        noInput: true,
+        sourceDir: neutrino.options.output
```

Comment:
If this directory contains any unimportant build artifacts then you should consider passing an [`ignoreFiles`](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/web-ext_command_reference#--ignore-files_-i) array to `run()`.

`run()` will start a file watcher within `sourceDir`. When a file gets touched, it will reload the extension unless the file matches any pattern in `ignoreFiles`.

It's important that the `web-ext` file watcher does *not* look for changes in the actual source files. It should only look for changes in the final executable files that are built from the source (I think that's how you have it configured). This is because if `web-ext` starts to reload the extension when a source file changes, it might reload too soon, before the final files are built by babel/webpack/whatever.